File: serial_comm/LogMessage.cpp
#include "LogMessage.h"

#include "Messages.h"
#include "Utils.h"

LogMessage::LogMessage(const String message) : MessageBase(Messages::LOG), payload(message)
{
  
}

Vector<unsigned char> LogMessage::encode()
{
  Vector<unsigned char> retVal(payload.length());

  for (unsigned int i = 0u; i < payload.length(); ++i)
  {
    retVal.push_back(payload.charAt(i));
  }

  return retVal;
}

void LogMessage::decode(const unsigned char* data, const uint16_t size)
{
  //Intentionally left empty
  UNUSED(data);
  UNUSED(size);
}

File: python/arduino_termpoint/comm_messaging/comm_serial/log_message.py
from ..message import Message
from .messages import Messages
import struct

class LogMessage(Message):
    def __init__(self):
        self.payload = ""

    def get_msg_type(self):
        return Messages.LOG

    def encode(self):
        return None

    def decode(self, data):
        self.payload = data.decode("utf-8")